### PR TITLE
Add Retry and Queue settings to honeycomb exporter

### PR DESCRIPTION
**Pull request by au-akash**
**Description:** 
Adding RetrySettings and QueueSettings to Honeycomb Exporter
Adding these settings offered by the exporterhelper will allow users of the Honeycomb exporter to retain queued retry behavior

**Testing:** 
RetrySettings and QueueSettings have been tested in exporterhelper, tests inside honeycomb exporter were changed to expect RetrySettings and QueueSettings

**Documentation:**
Added documentation on the new settings which can be configured on the exporter with descriptions on the defaults as well as their purpose

### CODE REVIEW

**Comment by bogdandrutu** [comment]
@paulosman @lizthegrey @MikeGoldsmith please review

**Comment by paulosman** [comment]
LGTM, thanks! @au-akash, you'll need to accept the CLA before the PR can be merged.

**Comment by au-akash** [comment]
Thanks @paulosman! We're working on getting CLA approval

**Comment by au-akash** [comment]
I signed it

**Comment by lizthegrey** [comment]
Please rebase/`commit --amend` in order to force checks to run again

**Comment by au-akash** [comment]
@lizthegrey I `commit --amend`, however the EasyCLA check is still failing.

This is how my linux foundation profile looks like:
image

**Comment by lizthegrey** [comment]
> @lizthegrey I `commit --amend`, however the EasyCLA check is still failing.
> 
> This is how my linux foundation profile looks like:
> image

can you screenshot what happens when you visit https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/9522512/196414933/2714/#/?version=1

**Comment by au-akash** [comment]
@lizthegrey Sorry for the delay, CLA has now been approved.

Feel free to merge whenever you're ready 🙏

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Based on @paulosman approver merging this

**Pull request merged**
